.github/workflows/x-release.yml

```diff
@@ -0,0 +1,107 @@
+name: X Release
+
+on:
+  workflow_call:
+    inputs:
+      gh-org:
+        required: true
+        type: string
+      branch:
+        required: true
+        type: string
+      tag:
+        required: true
+        type: string
+      nightly:
+        required: true
+        type: boolean
+    secrets:
+      NPM_TOKEN:
+        required: false
+
+concurrency: rel-${{ github.ref }}
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+
+  release-impl:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          check-latest: true
+          registry-url: https://registry.npmjs.org
+
+      - name: Create version commit
+        if: ${{ !inputs.nightly }}
+        run: |          
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "github-actions[bot]"
+          npm version ${{ inputs.tag }} -m "Set version to ${{ inputs.tag }}"          
+
+      - name: Tag commit
+        run: git tag ${{ inputs.tag }}
```

Comment:
By default `npm version` will tag the commit for us, so this step can be skipped for any version that is not the nightly.